### PR TITLE
11 delete comments

### PR DESCRIPTION
**Pull request by chris-butterworth**

### CODE REVIEW

File: src/components/Comments/CommentsCard.jsx [hunk 1]
```diff
@@ -17,8 +18,19 @@ export default function CommentCard({ comment, commentVotes }) {
       alert("your vote could not be added at this time");
     });
   };
+
+  const handleDeleteComment = (event) => {
+    event.preventDefault();
+    setHidden("hidden");
+    if (typeof comment.comment_id === "Number") {
```
Comment (chris-butterworth) [inline src/components/Comments/CommentsCard.jsx:25]: This 'Number' should be lower case 'number', I only spotted after I had merged.

**Review by chris-butterworth — COMMENTED** [review]
typo, can't work out how to change it once merged

File: src/components/Comments/CommentsCard.jsx (hunk 1 above)
Comment (Slyose) [reply src/components/Comments/CommentsCard.jsx:25]: Will have to change this on main then! This branch is already merged so we can't go back and add more code to it

**Comment by Slyose** [comment]
Work great for all comments apart from the optimistically rendered one - because this comment hasn't got the comment id from sql (that is used in a delete or vote request)

To get around this I'd either not optimistically render the comments, or find a workaround like not showing the delete/vote buttons on an optimistic comment.

**Comment by Slyose** [comment]
Also, this isn't on this branch but I noticed that when we vote on a comment and there's an error, the optimistic vote isnt taken back, which is the ideal behavior. Shouldn't be too hard to implement this - we just do the reverse of our optimistic vote in the .catch

**Pull request merged**